src/legendhpges/base.py

```diff
@@ -105,6 +110,65 @@ def _decode_polycone_coord(self) -> tuple[list[float], list[float]]:
         Must be overloaded by derived classes.
         """
 
+    def distance_to_surface(
+        self, coords: np.ndarray | list, surface_indices: list | None = None
```

Comment:
```suggestion
        self, coords: ArrayLike, surface_indices: ArrayLike | None = None
```
and similarly elsewhere.

See https://learn.scientific-python.org/development/guides/mypy/#loose-vs-specific-types and https://numpy.org/devdocs/reference/typing.html#numpy.typing.ArrayLike

Reply:
just fyi, typing support for array shapes is apparently still missing: https://github.com/numpy/numpy/issues/16544